### PR TITLE
Convert dbt_tags parameter to general select parameter

### PR DESCRIPTION
**Pull request by pohek321**
Closes [94](https://github.com/astronomer/astronomer-cosmos/issues/94)

I also snuck in there a max line length of 120 so that pre-commit stopped failing.

### CODE REVIEW

File: cosmos/providers/dbt/dag.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def __init__(
         emit_datasets: bool = True,
         dbt_root_path: str = "/usr/local/airflow/dbt",
         test_behavior: Literal["none", "after_each", "after_all"] = "after_each",
-        dbt_tags: List[str] = [],
+        select: Dict[str, Any] = {},
```
Comment (jlaneve) [inline cosmos/providers/dbt/dag.py:39]: could/should this be something like:

```suggestion
        select: Dict[str, List[str]] = {},
```

File: cosmos/providers/dbt/render.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def render_project(
     test_behavior: Literal["none", "after_each", "after_all"] = "after_each",
     emit_datasets: bool = True,
     conn_id: str = "default_conn_id",
-    dbt_tags: List[str] = [],
+    select: Dict = None,
```
Comment (jlaneve) [inline cosmos/providers/dbt/render.py:28]: could/should this be something like:
```suggestion
    select: Dict[str, List[str]] = None,
```

File: cosmos/providers/dbt/task_group.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ def __init__(
         emit_datasets: bool = True,
         dbt_root_path: str = "/usr/local/airflow/dbt",
         test_behavior: Literal["none", "after_each", "after_all"] = "after_each",
-        dbt_tags: List[str] = [],
+        select: Dict[str, Any] = {},
```
Comment (jlaneve) [inline cosmos/providers/dbt/task_group.py:39]: could/should this be something like:
```suggestion
        select: Dict[str, List[str]] = {},
```

Comment (pohek321) [reply cosmos/providers/dbt/task_group.py]: Added.

File: cosmos/providers/dbt/render.py (hunk 2 above)
Comment (pohek321) [reply cosmos/providers/dbt/render.py]: Added.

File: cosmos/providers/dbt/render.py [hunk 4]
```diff
@@ -56,8 +59,24 @@ def render_project(
...
-        if dbt_tags and not set(dbt_tags).intersection(model.config.tags):
-            continue
+
+        # ensures the same tag isn't in select & exclude
+        if "tags" in select and "tags" in exclude:
+            if set(select["tags"]).intersection(exclude["tags"]):
+                raise AirflowException(
+                    f"Can't specify the same tag in `select` and `include`: "
+                    f"{set(select['tags']).intersection(exclude['tags'])}"
+                )
```
Comment (jlaneve) [inline cosmos/providers/dbt/render.py:69]: we should be able to move this up, outside of the for loop for efficiency (the input tags won't change across each model)

**Review by jlaneve — APPROVED** [review]
just one small comment about where the exclude/select check is, but after that's addressed feel free to merge!

File: cosmos/providers/dbt/render.py (hunk 4 above)
Comment (pohek321) [reply cosmos/providers/dbt/render.py]: Done.

**Pull request merged**